packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -132,6 +132,17 @@ const emptyDispatchConfigForCustomEvents: CustomDispatchConfig = {
 
 const isArray = Array.isArray;
 
+// $FlowFixMe: Flow struggles with this pattern
+const PossiblyWeakMap = typeof WeakMap === 'function' ? WeakMap : Map;
+// $FlowFixMe: Flow cannot handle polymorphic WeakMaps
```

Comment:
Seems better to cast to `Map` (or `WeakMap`) so we have *some* type info rather than `$FlowFixMe`?

Reply:
I get the same issue with Maps, I kind of gave up here.